app/Jobs/SendSubscribedNotificationsJob.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Mail\CheckInNotificationMail;
+use App\Models\PhoneNumber;
+use App\Models\Subscription;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Mail;
+
+class SendSubscribedNotificationsJob implements ShouldQueue, ShouldBeUnique
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    public function __construct(
+        public PhoneNumber $phone_number,
+    )
+    {
+    }
+
+    public function uniqueId(): string
+    {
+        return $this->phone_number->id;
+    }
+
+    public function handle(): void
+    {
+        Cache::lock('send-notifications:'.$this->phone_number->id, 30)->get(function () {
```

Comment:
Would refactor `->id` to `->getKey()` but this looks great over all.

The only thing I don't quite follow is how the `30`-second lock here and the `120`-second delay when dispatching the job are intended to work together. Wouldn't these ideally be close to the same duration?

Reply:
These are solving 2 different issues. (Maybe a lack of understanding how "Unique Job" middleware works, but I was assuming that the max time it would take to send out the notifications is 30 seconds. 

If a Unique Job starts, then a new job dispatches before the first job ended, this was designed to prevent duplicate notifications

Reply:
"In the example above, the UpdateSearchIndex job is unique. So, the job will not be dispatched if another instance of the job is already on the queue **and has not finished processing.**" - [Docs](https://laravel.com/docs/11.x/queues#unique-jobs)


I think you are right and I can just remove the lock here altogether.